client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -215,6 +215,30 @@ private Checkpoint completeCheckpoint(String checkpointName) {
     public void resetReaderGroup(ReaderGroupConfig config) {
         log.info("Reset ReaderGroup {} to {}", getGroupName(), config);
         synchronizer.fetchUpdates();
+
+        //If null passed to ReaderGroupConfig then reset the reader group to last completed checkpoint.
+        if (config == null) {
+            val latestCheckpointConfig = synchronizer.getState().getConfig();
+            Optional<Map<Stream, Map<Segment, Long>>> lastCheckPointPositions =
+                    synchronizer.getState().getPositionsForLastCompletedCheckpoint();
+
+            Map<Stream, StreamCut> streamCuts = new HashMap<>();
+            if (lastCheckPointPositions.isPresent()) {
+                for (Entry<Stream, Map<Segment, Long>> streamPosition : lastCheckPointPositions.get().entrySet()) {
+                    streamCuts.put(streamPosition.getKey(), new StreamCutImpl(streamPosition.getKey(), streamPosition.getValue()));
+                }
+
+            } else {
+                log.info("Reset reader group to last completed checkpoint is not successful as there is no checkpoint available, so resetting to start of the stream ");
+                Set<String> streamNames = getStreamNames();
+                for (String stream : streamNames) {
+                    streamCuts.put(Stream.of(stream), StreamCut.UNBOUNDED);
+                }
+            }
+            //apply streamcut to reader group config from reader group state object
+            ReaderGroupConfig readerConfig = latestCheckpointConfig.toBuilder().startingStreamCuts(streamCuts).build();
+            config = ReaderGroupConfig.cloneConfig(readerConfig, latestCheckpointConfig.getReaderGroupId(), latestCheckpointConfig.getGeneration());
```

Comment:
I am not sure what this line is supposed to be doing. `readerConfig` is a config which is derived from `latestCheckpointConfig` and only differs in the `startingStreamCuts` field. Here it appears to copy the config and assign the groupId and generation from latestCheckpointConfig which is redundant because they already are these values. Then the copy is assigned to `config` and `readerConfig` is dropped. So this whole line is currently equivalent to `config = readerConfig`. But it seems like it was intended to do something else.

Reply:
Intend was to reconstruct the config by applying startingStreamCuts to latestCheckpointConfig. But you are right as we are using groupId and generation from latestCheckpointConfig again getting clone of same is not required. Will Do the necessary changes.